File: SDKs/Aspose.Tasks_Cloud_SDK_for_Ruby/lib/aspose_tasks_cloud/models/extended_attribute_definition.rb
module AsposeTasksCloud
  # 
  class ExtendedAttributeDefinition < BaseObject
    attr_accessor :field_id, :field_name, :cf_type, :guid, :element_type, :max_multi_values, :user_def, :_alias, :secondary_pid, :auto_roll_down, :default_guid, :lookup_uid, :phonetics_alias, :rollup_type, :calculation_type, :formula, :restrict_values, :valuelist_sort_order, :append_new_values, :default, :value_list, :secondary_guid
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # 
        :'field_id' => :'FieldId',
        
        # 
        :'field_name' => :'FieldName',
        
        # 
        :'cf_type' => :'CfType',
        
        # 
        :'guid' => :'Guid',
        
        # 
        :'element_type' => :'ElementType',
        
        # 
        :'max_multi_values' => :'MaxMultiValues',
        
        # 
        :'user_def' => :'UserDef',
        
        # 
        :'_alias' => :'Alias',
        
        # 
        :'secondary_pid' => :'SecondaryPid',
        
        # 
        :'auto_roll_down' => :'AutoRollDown',
        
        # 
        :'default_guid' => :'DefaultGuid',
        
        # 
        :'lookup_uid' => :'LookupUid',
        
        # 
        :'phonetics_alias' => :'PhoneticsAlias',
        
        # 
        :'rollup_type' => :'RollupType',
        
        # 
        :'calculation_type' => :'CalculationType',
        
        # 
        :'formula' => :'Formula',
        
        # 
        :'restrict_values' => :'RestrictValues',
        
        # 
        :'valuelist_sort_order' => :'ValuelistSortOrder',
        
        # 
        :'append_new_values' => :'AppendNewValues',
        
        # 
        :'default' => :'Default',
        
        # 
        :'value_list' => :'ValueList',
        
        # 
        :'secondary_guid' => :'SecondaryGuid'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'field_id' => :'String',
        :'field_name' => :'String',
        :'cf_type' => :'String',
        :'guid' => :'String',
        :'element_type' => :'String',
        :'max_multi_values' => :'Integer',
        :'user_def' => :'BOOLEAN',
        :'_alias' => :'String',
        :'secondary_pid' => :'String',
        :'auto_roll_down' => :'BOOLEAN',
        :'default_guid' => :'String',
        :'lookup_uid' => :'String',
        :'phonetics_alias' => :'String',
        :'rollup_type' => :'String',
        :'calculation_type' => :'String',
        :'formula' => :'String',
        :'restrict_values' => :'BOOLEAN',
        :'valuelist_sort_order' => :'Integer',
        :'append_new_values' => :'BOOLEAN',
        :'default' => :'String',
        :'value_list' => :'Array<Value>',
        :'secondary_guid' => :'String'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'FieldId']
        self.field_id = attributes[:'FieldId']
      end
      
      if attributes[:'FieldName']
        self.field_name = attributes[:'FieldName']
      end
      
      if attributes[:'CfType']
        self.cf_type = attributes[:'CfType']
      end
      
      if attributes[:'Guid']
        self.guid = attributes[:'Guid']
      end
      
      if attributes[:'ElementType']
        self.element_type = attributes[:'ElementType']
      end
      
      if attributes[:'MaxMultiValues']
        self.max_multi_values = attributes[:'MaxMultiValues']
      end
      
      if attributes[:'UserDef']
        self.user_def = attributes[:'UserDef']
      end
      
      if attributes[:'Alias']
        self._alias = attributes[:'Alias']
      end
      
      if attributes[:'SecondaryPid']
        self.secondary_pid = attributes[:'SecondaryPid']
      end
      
      if attributes[:'AutoRollDown']
        self.auto_roll_down = attributes[:'AutoRollDown']
      end
      
      if attributes[:'DefaultGuid']
        self.default_guid = attributes[:'DefaultGuid']
      end
      
      if attributes[:'LookupUid']
        self.lookup_uid = attributes[:'LookupUid']
      end
      
      if attributes[:'PhoneticsAlias']
        self.phonetics_alias = attributes[:'PhoneticsAlias']
      end
      
      if attributes[:'RollupType']
        self.rollup_type = attributes[:'RollupType']
      end
      
      if attributes[:'CalculationType']
        self.calculation_type = attributes[:'CalculationType']
      end
      
      if attributes[:'Formula']
        self.formula = attributes[:'Formula']
      end
      
      if attributes[:'RestrictValues']
        self.restrict_values = attributes[:'RestrictValues']
      end
      
      if attributes[:'ValuelistSortOrder']
        self.valuelist_sort_order = attributes[:'ValuelistSortOrder']
      end
      
      if attributes[:'AppendNewValues']
        self.append_new_values = attributes[:'AppendNewValues']
      end
      
      if attributes[:'Default']
        self.default = attributes[:'Default']
      end
      
      if attributes[:'ValueList']
        if (value = attributes[:'ValueList']).is_a?(Array)
          self.value_list = value
        end
      end
      
      if attributes[:'SecondaryGuid']
        self.secondary_guid = attributes[:'SecondaryGuid']
      end
      
    end

  end
end

File: SDKs/Aspose.Tasks-Cloud-SDK-for-.NET/src/Com/Aspose/Tasks/Model/ExtendedAttributeDefinition.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace Com.Aspose.Tasks.Model {
  public class ExtendedAttributeDefinition {
    public string FieldId { get; set; }

    public string FieldName { get; set; }

    public CustomFieldType CfType { get; set; }

    public string Guid { get; set; }

    public ElementType ElementType { get; set; }

    public int? MaxMultiValues { get; set; }

    public bool? UserDef { get; set; }

    public string Alias { get; set; }

    public string SecondaryPid { get; set; }

    public bool? AutoRollDown { get; set; }

    public string DefaultGuid { get; set; }

    public string LookupUid { get; set; }

    public string PhoneticsAlias { get; set; }

    public RollupType RollupType { get; set; }

    public CalculationType CalculationType { get; set; }

    public string Formula { get; set; }

    public bool? RestrictValues { get; set; }

    public int? ValuelistSortOrder { get; set; }

    public bool? AppendNewValues { get; set; }

    public string Default { get; set; }

    public List<Value> ValueList { get; set; }

    public string SecondaryGuid { get; set; }

    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class ExtendedAttributeDefinition {\n");
      sb.Append("  FieldId: ").Append(FieldId).Append("\n");
      sb.Append("  FieldName: ").Append(FieldName).Append("\n");
      sb.Append("  CfType: ").Append(CfType).Append("\n");
      sb.Append("  Guid: ").Append(Guid).Append("\n");
      sb.Append("  ElementType: ").Append(ElementType).Append("\n");
      sb.Append("  MaxMultiValues: ").Append(MaxMultiValues).Append("\n");
      sb.Append("  UserDef: ").Append(UserDef).Append("\n");
      sb.Append("  Alias: ").Append(Alias).Append("\n");
      sb.Append("  SecondaryPid: ").Append(SecondaryPid).Append("\n");
      sb.Append("  AutoRollDown: ").Append(AutoRollDown).Append("\n");
      sb.Append("  DefaultGuid: ").Append(DefaultGuid).Append("\n");
      sb.Append("  LookupUid: ").Append(LookupUid).Append("\n");
      sb.Append("  PhoneticsAlias: ").Append(PhoneticsAlias).Append("\n");
      sb.Append("  RollupType: ").Append(RollupType).Append("\n");
      sb.Append("  CalculationType: ").Append(CalculationType).Append("\n");
      sb.Append("  Formula: ").Append(Formula).Append("\n");
      sb.Append("  RestrictValues: ").Append(RestrictValues).Append("\n");
      sb.Append("  ValuelistSortOrder: ").Append(ValuelistSortOrder).Append("\n");
      sb.Append("  AppendNewValues: ").Append(AppendNewValues).Append("\n");
      sb.Append("  Default: ").Append(Default).Append("\n");
      sb.Append("  ValueList: ").Append(ValueList).Append("\n");
      sb.Append("  SecondaryGuid: ").Append(SecondaryGuid).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  }
